assets/src/editor/events/initialization/editor-initialization.js

```diff
@@ -0,0 +1,40 @@
+import { useEffect } from '@wordpress/element';
+import { compose, ifCondition } from '@wordpress/compose';
+
+import useEditorInitialization from './use-editor-initialization';
+import withEventEntity from './with-event-entity';
+import withEventDateEntities from './with-event-date-entities';
+import withTicketEntitiesForAllDateEntities from './with-ticket-entities-for-all-date-entities';
+import { Provider } from './context';
+
+const EditorInitialization = ( {
+	eventEntityLoaded,
+	dateEntitiesLoaded,
+	ticketEntitiesLoaded,
+	editorInitialized,
+	setInitialized,
+	children,
+} ) => {
+	useEffect( () => {
+		if ( eventEntityLoaded && dateEntitiesLoaded && ticketEntitiesLoaded ) {
+			setInitialized( true );
+		}
+	}, [ eventEntityLoaded, dateEntitiesLoaded, ticketEntitiesLoaded ] );
+	return <Provider value={ editorInitialized }>
+		{ children }
+	</Provider>;
```

Comment:
Essentially the `EditorInitialization` component (utilized via the `withEditorInitialization` hoc) wraps children in a Context Provider exposing the initialization state.  You can see here that currently the editor is considered "initialized" when the eventEntity, related dates to that event, and related tickets to those dates are all retrieved from the server and added to the state.  If we want to change what is exposed on the initialization context (or what criteria makes something "initialized" this is where we'd do it).